templates/helperHeader.dot

```diff
@@ -11,9 +11,63 @@
 #define EXTRACT_v8_String(x) EXTRACT_v8_string(x)
 #define EXTRACT_v8_float(x) EXTRACT_v8_double(x)
 
-struct ArrayBuffer {
-  char* data;
-  uint32_t size;
+class ArrayBufferHelper {
+ public:
+  ArrayBufferHelper(void* data, uint32_t length) {
+    Set(data, length);
+  }
+
+  ArrayBufferHelper(const ArrayBufferHelper& rhs) {
+    CopyFrom(rhs);
+  }
+
+  ~ArrayBufferHelper() {}
+
+  ArrayBufferHelper& operator = (const ArrayBufferHelper& rhs) {
+    CopyFrom(rhs);
+    return *this;
+  }
+
+  void Set(void* data, uint32_t length, bool external = true) {
+    data_ = data;
+    length_ = length;
+    external_buffer_ = external;
+  }
+
+  v8::Local<v8::ArrayBuffer> GetArrayBuffer() const {
+    auto mode = external_buffer_ ?
+        v8::ArrayBufferCreationMode::kExternalized :
+        v8::ArrayBufferCreationMode::kInternalized;
+
+    return v8::ArrayBuffer::New(
+        v8::Isolate::GetCurrent(), data_, length_, mode);
+  }
+
+  bool operator == (const ArrayBufferHelper& rhs) {
+    if (this != &rhs) {
+      return data_ == rhs.data_ &&
+          length_ == rhs.length_ &&
+          external_buffer_ == rhs.external_buffer_;
+    }
+    return true;
+  }
+
+  void* GetData() const { return data_; }
+  uint32_t GetLength() const { return length_; }
+  bool IsExternalBuffer() const { return external_buffer_; }
+
+ private:
+  void CopyFrom(const ArrayBufferHelper& rhs) {
+    if (this != &rhs) {
+      data_ = rhs.data_;
```

Comment:
How about we name this one `ExternalArrayBuffer` and make it only work in `external buffer mode`?

`ArrayBuffer`s with ownership of buffer will be treated in other class, to avoid potential issues.